azkaban-common/src/main/java/azkaban/executor/AbstractExecutorManagerAdapter.java

```diff
@@ -341,6 +352,11 @@ protected String uploadExecutableFlow(
     // The exflow id is set by the loader. So it's unavailable until after
     // this call.
     this.executorLoader.uploadExecutableFlow(exflow);
+    // Emit dispatching flow event
+    exflow.setStatus(Status.DISPATCHING);
+    final PodEventListener podEventListener = new PodEventListener();
```

Comment:
Can you please use fireEventListeners? With that, fireEventListeners can have one parameter which will be exflow.
With this, as it will be status changed event, please rename fireEventListeners to fireStatusChangeEvent or something?

Reply:
Used eventhandler to fire event listener.